packages/babel-traverse/src/path/lib/removal-hooks.js

```diff
@@ -64,5 +64,18 @@ export let hooks = [
       }
       return true;
     }
+  },
+
+  function (self, parent) {
+    if (
+      (self.key === 'consequent' || self.key === 'alternate') && parent.isIfStatement() ||
```

Comment:
Seems like we should check the parent type first to be consistent with the rest of these, and to help readability. I'd also be in favor of adding more parens around these `&&` vs `||` groups since I never remember their precedence.